File: src/main/java/com/aladdinworks33/service/NetworkAlertService.java
package com.aladdinworks33.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.aladdinworks33.domain.NetworkAlert;
import com.aladdinworks33.dto.NetworkAlertDTO;
import com.aladdinworks33.dto.NetworkAlertSearchDTO;
import com.aladdinworks33.dto.NetworkAlertPageDTO;
import com.aladdinworks33.dto.NetworkAlertConvertCriteriaDTO;
import com.aladdinworks33.service.GenericService;
import com.aladdinworks33.dto.common.RequestDTO;
import com.aladdinworks33.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface NetworkAlertService extends GenericService<NetworkAlert, Integer> {

	List<NetworkAlert> findAll();

	ResultDTO addNetworkAlert(NetworkAlertDTO networkAlertDTO, RequestDTO requestDTO);

	ResultDTO updateNetworkAlert(NetworkAlertDTO networkAlertDTO, RequestDTO requestDTO);

    Page<NetworkAlert> getAllNetworkAlerts(Pageable pageable);

    Page<NetworkAlert> getAllNetworkAlerts(Specification<NetworkAlert> spec, Pageable pageable);

	ResponseEntity<NetworkAlertPageDTO> getNetworkAlerts(NetworkAlertSearchDTO networkAlertSearchDTO);
	
	List<NetworkAlertDTO> convertNetworkAlertsToNetworkAlertDTOs(List<NetworkAlert> networkAlerts, NetworkAlertConvertCriteriaDTO convertCriteria);

	NetworkAlertDTO getNetworkAlertDTOById(Integer networkAlertId);







}

File: aladdinworks33_ui/src/services/NetworkAlertService.js
import http from "../http-common"; 

class NetworkAlertService {
  getAllNetworkAlerts(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/networkAlert/networkAlerts`, searchDTO);
  }

  get(networkAlertId) {
    return this.getRequest(`/networkAlert/${networkAlertId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/networkAlert?field=${matchData}`, null);
  }

  addNetworkAlert(data) {
    return http.post("/networkAlert/addNetworkAlert", data);
  }

  update(data) {
  	return http.post("/networkAlert/updateNetworkAlert", data);
  }
  
  uploadImage(data,networkAlertId) {
  	return http.postForm("/networkAlert/uploadImage/"+networkAlertId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new NetworkAlertService();
